### DISCUSSION TITLE
Access Asset Path

### DISCUSSION THREAD

**Discussion by BinaryAssault** [dotnet/maui#7899] [Q&A]
Say I have an asset called `livefree.ogg` that is located in `Raw/Songs`
```
Application
  |-Resources
    |-Raw
      |-Songs
        |-livefree.ogg -> Build Action: Maui Asset
```

If I need to say, loop over all the files in Raw/Songs, how do I do so? Using `FileSystem.OpenAppPackageFileAsync` doesn't get me access to the path of the item, just a stream. Which in my case, I need a path and not a stream. Is this possible with MAUI?

None of the following return any files:
``````
Directory.GetFiles("Songs"); 
Directory.GetFiles("/Songs"); 
Directory.GetFiles("Resources/Songs");
Directory.GetFiles("/Resources/Songs");
Directory.GetFiles("Raw/Songs");
Directory.GetFiles("/Raw/Songs"); 
Directory.GetFiles("Resources/Raw/Songs"); 
Directory.GetFiles("/Resources/Raw/Songs"); 
Directory.GetFiles(Path.Combine(FileSystem.AppDataDirectory, "Songs")); 
Directory.GetFiles(Path.Combine(FileSystem.AppDataDirectory, "Raw", "Songs"));
Directory.GetFiles(Path.Combine(FileSystem.AppDataDirectory, "Resource", "Raw", "Songs"));
``````

**Comment by u1-kino** [comment]
[MauiApp1.zip](https://github.com/dotnet/maui/files/9609020/MauiApp1.zip)

The attached sample app successfully reads files from a folder within the app. Operation has been confirmed with Visual Studio 2022 for Windows.

- Project settings
It is necessary to change the project settings for iOS and MacCatalyst apps in VisualStudio2022 for Mac.

- Source code
The file loading process is different for each platform.

I don't know if it will be helpful, but please check the operation of the sample application.
@BinaryAssault

**Comment by maharasa** [comment]
Hello,

I'm seeking assistance regarding accessing an SQLite file in my Blazor WebAssembly application. Previously, I was able to access the file using the AppDomain.CurrentDomain.BaseDirectory +"wwwroot", "Data", "MDA.db" path and connect to it using Entity Framework Core. However, I have recently converted my application to MAUI Blazor and I want to continue accessing the SQLite file from the same path.

I have tried using the following code to access the file:
var path = Path.Combine(FileSystem.Current.AppDataDirectory, "wwwroot", "Data", "MDA.db");

However, I am encountering an SQLite Error 14: 'unable to open database file'.

I have also attempted using the following path instead:
var path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "wwwroot", "Data", "MDA.db");

This works fine during development, but when installed on a client machine, it fails to work.

Could someone please suggest a solution for this issue?
